src/Build/Evaluation/Profiler/EvaluationLocationTabSeparatedPrettyPrinter.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//-----------------------------------------------------------------------
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Text;
+using Microsoft.Build.Framework.Profiler;
+
+namespace Microsoft.Build.Evaluation
+{
+    /// <summary>
+    /// Pretty prints an evaluation location in tab separated value (TSV) format
+    /// </summary>
+    internal sealed class EvaluationLocationTabSeparatedPrettyPrinter : EvaluationLocationPrettyPrinterBase
+    {
+        /// <inheritdoc/> 
+        internal override void AppendHeader(StringBuilder stringBuilder)
+        {
+            stringBuilder.AppendLine("Id\tParentId\tPass\tFile\tLine #\tExpression\tInc (ms)\tInc (%)\tExc (ms)\tExc (%)\t#\tKind\tBug");
+        }
+
+        /// <inheritdoc/>
+        internal override void AppendLocation(StringBuilder stringBuilder, TimeSpan totalTime, EvaluationLocation evaluationLocation, ProfiledLocation profiledLocation)
+        {
+            stringBuilder.AppendLine(string.Join("\t",
+                evaluationLocation.Id,
+                evaluationLocation.ParentId?.ToString() ?? string.Empty,
+                evaluationLocation.EvaluationPassDescription,
```

Comment:
Nit: row data can be pulled up in the superclass, and the base classes provide separator / string printing

Reply:
We could, but not sure if it is worth the effort specializing it like that... I could add a helper method on the base class, sounds like less coupling

---
In reply to: [159566570](https://github.com/Microsoft/msbuild/pull/2837#discussion_r159566570) [](ancestors = 159566570)